custom_components/waste_collection_schedule/waste_collection_schedule/source/egn_abfallkalender_de.py

```diff
@@ -0,0 +1,91 @@
+import datetime
+import urllib
+import logging
+from typing import Dict
+
+import requests
+from bs4 import BeautifulSoup
+from requests.sessions import dispatch_hook
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "EGN Abfallkalender"
+DESCRIPTION = "Source for EGN Abfallkalender"
+URL = "https://www.egn-abfallkalender.de/kalender"
+TEST_CASES: Dict[str, Dict[str, str]] = {
```

Comment:
I'm not very familiar with typing, but the dict value you use for `days` is not a str. Is this data typing correct, therefore? And it would be nice if housenumber could be also specified as an int.